internal/jujucommands/jujucommands_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2025 Canonical.
+
+package jujucommands_test
+
+import (
+	"strings"
+	"testing"
+
+	qt "github.com/frankban/quicktest"
+	"github.com/frankban/quicktest/qtsuite"
+	"github.com/juju/juju/jujuclient"
+
+	"github.com/canonical/jimm/v3/internal/jujucommands"
+)
+
+type jujucommandsSuite struct{}
+
+func (s *jujucommandsSuite) TestRunCmdWithOutputRetriever(c *qt.C) {
+	outputCh, err := jujucommands.RunCmdWithOutputRetriever(jujuclient.NewEmbeddedMemStore(), "help")
+	c.Assert(err, qt.IsNil)
+
+	// Use a builder to collect streamed output & test entire string.
+	var b strings.Builder
+
+	for out := range outputCh {
+		c.Assert(out.Err, qt.IsNil)
+		b.WriteString(out.Line + "\n")
```

Comment:
It wasn't documented above that the lines from the output channel will not have a new-line character. I suggest adding that fact to the docstring.